packages/cli/src/commands/deploy/helper.ts

```diff
@@ -77,18 +87,30 @@ export const upsertFieldPlugin: UpsertFieldPluginFunc = async (args) => {
     // update flow
     const mode = skipPrompts ? 'update' : await selectUpsertMode()
     if (mode === 'update') {
+      if (manifest?.options !== undefined) {
+        const confirmed = await confirmOptionsUpdate()
+
+        if (!confirmed) {
+          console.log(cyan(bold('[info]')), 'Aborting plugin update.')
+          process.exit(1)
+        }
+      }
+
       await storyblokClient.updateFieldType({
         id: fieldPlugin.id,
         field_type: {
           body: output,
+          options: manifest?.options,
```

Comment:
I'm not assuming any default value for `manifest?.options` but `undefined`.

This way, in case no manifest file was found, options defined explicitly by the user (through the plugin visual editor) are not going to be erased by a default value.

**Scenario**:

1. The user creates and deploys a new field plugin without any manifest file options
2. Manually the user creates options directly through the Plugin Visual Editor
3. The user update the plugin using the CLI
4. The CLI assumes a default empty array for the options (in case no manifest file is found or no options are defined on it) and replaces the options manually added (second step) with this empty array.

We're going only to update the options if a valid manifest file is found, otherwise, it'll be undefined and not be sent to the API, preserving the manually added ones.

This approach is going to avoid erasing options by mistake when no valid manifest file is found.

However, if a user adds some options manually through the Plugin Visual Editor and after creating a manifest file with an empty array, the previous options are going to be erased, but first, the user is going to be prompted and asked to confirm it, so it'll be safe.

Am I forgetting some aspect here? Is there some pitfalls in this approach I'm not seeing? 🤔

Reply:
> why is manifest?.options optionally chained if you already checked it in the previous if statement? shouldn't it be just manifest.options?

Hey @BibiSebi, if I understood you correctly, it was checked before but in another block, here in line 102, it still could be undefined (a situation where no manifest file was defined, for example).

Reply:
> I have a question. Have you already tested this API? Let's assume there is a field plugin, and I manually added some options on Storyblok. I wonder if I call the following code, will it erase the existing options or not?

Hey @eunjae-lee, it won't erase the options in that case. 
The options manually added will still be preserved. 🙌 

Since the `options list` will be `undefined` it'll not be sent to the API because `undefined` is not a valid JSON and it will be removed before the request is made.

![image](https://github.com/storyblok/field-plugin/assets/1240591/84fe379d-5c6a-440c-8989-9c775a70edf6)

So, if no valid manifest file is provided, not `options` list will be sent to the API and it will work as it worked before, preserving the options manually added.